kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs

```diff
@@ -1367,6 +1380,52 @@ impl<'tcx> GotocCtx<'tcx> {
             _ => false,
         }
     }
+
+    /// Generate code for a valid object-safe trait receiver type.
+    ///
+    /// Note that all these types only contain the data pointer and ZST fields. Thus, we generate
+    /// the non-ZST branch manually. In some cases, this method is called from inside
+    /// `codegen_ty(arg_ty)` so we don't have information about the final type.
```

Comment:
If this is a problem, tracking issue to fix this?

Reply:
This is just a note that if the language changes, this might not work.